### ISSUE TITLE
Flamegraph: Different modes of color coding

### ISSUE THREAD

**Issue by aocenas** [grafana/grafana#57673]
### Value color coding
At this moment we color code based on the Value of each of the bars. This means that bigger the bar the more red it will be. This creates something like this:
<img src="https://user-images.githubusercontent.com/1014802/198037894-b615c90c-85ba-414c-a574-e49e17edf893.png" width=400/>

This sort of duplicates information that is already available (the size of the bars) but can still be valuable to be able to quickly drill down into the part of the graph that is bigger.

### Self color coding
Another approach would be to color based on Self value. This would give a quick insight into which function call was actually doing the most work or allocated the most memory, possibly surfacing actually bottlenecks more quickly.
<img src="https://user-images.githubusercontent.com/1014802/198039135-02a243be-9926-4766-9c98-30738d9321b8.png" width=400/>

### Code type color coding
Third possibility is to color based on the name of the bar. This could either color in a way that similar names have similar color (sort of like trace view colors the same services the same way) or based on the language in case of multilanguage profiles so that for example Java calls have one color scheme and system calls different one (see https://www.brendangregg.com/blog/2017-07-30/coloring-flamegraphs-code-type.html).

### tldr
In summary it seems like different color coding may be useful for different usecases and having a switch so that user can decide on the coloring strategy would be the best approach.

**Comment by aocenas** [comment]
Draft PR for the self value color coding https://github.com/grafana/grafana/pull/57674

**Comment by aocenas** [comment]
We already implemented value and code color coding, the self color scheme does not seem like it adds that much value right now.
